src/htr2hpc/tasks.py

```diff
@@ -0,0 +1,230 @@
+    # HOWEVER: when a model name is specified, the form process method
+    # creates a new, empty model db record with that name; when override
+    # is not requested, form logic creates a copy of the model.
+    # So we may need to revise the script with an option to update
+    # the specified model.
+
+    # we require both of these; are they really optional?
+    if not all([user_pk, document_pk]):
+        # can't proceed without both of these
+        logger.error(
+            f"segtrain called without document_pk or user_pk; document_pk={document_pk} user_pk={user_pk}"
+        )
+        return
+
+    try:
+        user = User.objects.get(pk=user_pk)
+    except User.DoesNotExist:
+        # error / bail out
+        logger.error(f"segtrain called with invalid user_pk {user_pk}")
+        return
+
+    # get the requested model from the db
+    OcrModel = apps.get_model("core", "OcrModel")
+    model = OcrModel.objects.get(pk=model_pk)
+    # mark the model as being in training
+    # would be nice if the script could handle, but that field is listed
+    # as read only in the api
+    model.training = True
+    model.save()
+    # send event to indicate training is starting
+    send_event(
+        "document",
+        document_pk,
+        "training:start",
+        {
+            "id": model_pk,
+        },
+    )
+
+    # assume we're using LDAP accounts only so usernames match here and on hpc
+    username = user.username
+    api_token = user.auth_token.key
+
+    # create a name for an output directory based on mode and document id
+    working_dir = f"/scratch/gpfs/{username}/htr2hpc"
+    # TODO: should we add a timestamp to ensure uniqueness?
+    # script will fail if there is an existing directory
+    outdir = f"segtrain_doc{document_pk}"
+
+    # generate the command to run
+    site = Site.objects.get(pk=settings.SITE_ID)
+    site_url = site.domain
+    if not site_url.startswith("http"):
+        site_url = f"https://{site_url}"
+
+    arg_options = [
+        f"--document {document_pk}",  # document id is always required
+        f"--model-name {model.name}",
+        "--no-progress",  # disable progressbar
+    ]
+
+    # part and model are optional
+    if part_pks:
+        # parse and serialize with intspan since that's what we use on the other side
+        arg_options.append(f"--parts {intspan(part_pks)}")
+    if model_pk:
+        arg_options.append(f"--model {model_pk}")
+    opts = " ".join(arg_options)
+
+    cmd = f"htr2hpc-train segmentation {site_url} {outdir} {opts}"
+    # for now just output the command
+    logger.info(cmd)
+
+    # hostname and ssh key path set in django config
+    logger.debug(
+        f"Connecting to {settings.HPC_HOSTNAME} as {username} with keyfile {settings.HPC_SSH_KEYFILE}"
+    )
+
+    user.notify(
+        "Starting remote training; slurm portion can be monitored on mydella",
+        links=["https://mydella.princeton.edu/pun/sys/dashboard/activejobs"],
```

Comment:
```suggestion
        links=[{'text': 'Della Active Jobs', 'src': "https://mydella.princeton.edu/pun/sys/dashboard/activejobs"}],
```

Based on other examples of `user.notify` in the eScr code, this might be the format needed for links?

Reply:
thank you! I didn't have the chance to go digging for other examples yet